Atomic/backends/_docker.py

```diff
@@ -0,0 +1,271 @@
+            return con_obj
+        return None
+
+    def _inspect_image(self, image):
+        try:
+            inspect_data = self.d.inspect_image(image)
+        except errors.NotFound:
+            return None
+        return inspect_data
+
+    def inspect_image(self, image):
+        inspect_data = self._inspect_image(image)
+        if inspect_data:
+            img_obj = self._make_image(image, inspect_data, deep=True)
+            return img_obj
+        return None
+
+    def _make_image(self, image, img_struct, deep=False, remote=False):
+        img_obj = Image(image, remote=remote)
+        if not remote:
+            img_obj.id = img_struct['Id']
+            img_obj.backend = self
+            img_obj.repotags = img_struct['RepoTags']
+            img_obj.created = img_struct['Created']
+            img_obj.size = img_struct['Size']
+
+        if deep:
+            img_obj.deep = True
+            img_obj.repotags = img_struct['RepoTags']
+            img_obj.config = img_struct['Config'] or {}
+            img_obj.labels = img_obj.config.get("Labels", None)
+            img_obj.os = img_struct['Os']
+            img_obj.arch = img_struct['Architecture']
+            img_obj.graph_driver = img_struct['GraphDriver']
+            img_obj.version = img_obj.get_label('Version')
+            img_obj.release = img_obj.get_label('Release')
+            img_obj.parent = img_struct['Parent']
+        return img_obj
+
+    def _make_container(self, container, con_struct, deep=False):
+        con_obj = Container(container)
+        con_obj.id = con_struct['Id']
+        con_obj.created = con_struct['Created']
+        con_obj.original_structure = con_struct
+        con_obj.input_name = container
+        con_obj.backend = self
+
+        if not deep:
+            con_obj.status = con_struct['Status']
+            con_obj.image = con_struct['ImageID']
+
+        if deep:
+            # Add in the deep inspection stuff
+            con_obj.status = con_struct['State']['Status']
+            con_obj.running = con_struct['State']['Running']
+            con_obj.image = con_struct['Image']
+
+
+        return con_obj
+
+    def _inspect_container(self, container):
+        try:
+            inspect_data = self.d.inspect_container(container)
+        except errors.NotFound:
+            return None
+        return inspect_data
+
+    def inspect_container(self, container):
+        inspect_data = self._inspect_container(container)
+        if inspect_data:
+            return self._make_container(container, inspect_data, deep=True)
+        return None
+
+    def get_images(self, get_all=False):
+        images = self.get_docker_images(get_all=get_all)
+        image_objects = []
+        for image in images:
+            image_objects.append(self._make_image(image['Id'], image))
+        return image_objects
+
+    def make_remote_image(self, image):
+        img_obj = self._make_remote_image(image)
+        img_obj.remote_inspect()
```

Comment:
My point is remote_inspect is a horrible name for this function.  I would figure this would  return a json file describing the image.  Why does it exist as a public member, Why is it not built into _make_remote_image?